### PR TITLE
Fixed initial ewm phase always being zero

### PR DESCRIPTION
**Pull request by bonventre**
Fixed a typo that made the initial phase always zero. This PR will change the nightly validation slightly again since the event window times in the 10 validation runs will be different.

### CODE REVIEW

File: CommonMC/src/EventWindowMarkerProducer_module.cc [hunk 1]
```diff
@@ -38,6 +38,8 @@ namespace mu2e {
         fhicl::Atom<unsigned> spillType { Name( "SpillType"), Comment("Whether simulating on or off spill")};
         fhicl::Atom<bool> recoFromMCTruth{ Name("RecoFromMCTruth"), Comment("Create reco PBT from MC truth"),false};
         fhicl::Atom<float> recoFromMCTruthErr{ Name("RecoFromMCTruthErr"), Comment("If creating reco PBT from MC truth, smear by this amount in ns"),0};
+        fhicl::Atom<bool> fixInitialPhase{ Name("FixInitialPhase"), Comment("If false will randomize phase every run"),false};
+        fhicl::Atom<float> initialPhaseShift{ Name("InitialPhaseShift"), Comment("Initial phase shift between beam and detector clocks in fraction of one clock tick."),0};
```
Comment (brownd1978) [inline CommonMC/src/EventWindowMarkerProducer_module.cc:42]: Andrei has pointed out a better design pattern for this kind of config, namely OptionalAtom.   In this case initialPhaseShift would become an OptionalAtom, and the bool Atom would go away.  The test on the bool would be replaced by a test whether the OptionalAtom is set or not.  This combines the config and prevents ambiguous situations (ie what to do if someone sets fixInitialPhase to false but doesn't provide a value, etc).  An example is

**Review by brownd1978 — CHANGES_REQUESTED** [review]

File: CommonMC/src/EventWindowMarkerProducer_module.cc (hunk 1 above)
Comment (bonventre) [reply CommonMC/src/EventWindowMarkerProducer_module.cc:42]: That's a lot more convenient, good to know that exists!

**Review by brownd1978 — APPROVED** [review]
looks good.  Should there be a production configuration for this for MDC2020.

**Comment by brownd1978** [comment]
@FNALbuild run build test

**Comment by kutschke** [comment]
@FNALbuild run build tests

**Comment by gaponenko** [comment]
Hi,

> Andrei has pointed out a better design pattern for this kind of config, namely OptionalAtom.   In this case initialPhaseShift would become an OptionalAtom, and the bool Atom would go away.  The test on the bool would be replaced by a test whether the OptionalAtom is set or not.  This combines the config and prevents ambiguous situations (ie what to do if someone sets fixInitialPhase to false but doesn't provide a value, etc).  An example is 

This reminded me that there is an esier-to-use interface for Optional
entities.  See https://cdcvs.fnal.gov/redmine/issues/25773
It looks like it should be available in art v3.09.00; I have not tried it yet.

Andrei

**Comment by brownd1978** [comment]
@FNALbuild run build tests

**Comment by brownd1978** [comment]
@FNALbuild run build tests

**Pull request merged**